tests/test_cli.py

```diff
@@ -21,24 +24,113 @@
 # TODO simulate container runtime missing
 # TODO simulate container connection error
 # TODO simulate container connection loss
-# FIXME "/" path separator is platform-dependent, use pathlib instead
+
+
+class CLIResult:
```

Comment:
We may want to check if something is in the output of the run command with `Result.stdout` or some other `Result` method. The way `CliResult` was is implemented here it wraps around the `click.Result` object but prevented calling its methods. I'd propose making this a subclass of Result.  I have a proposal at `origin/234-windows-unit-tests-2`. It looks a bit hacky but it works.

Reply:
This approach has the extra advantage of not having to call `self.click_result` but instead `self`.

Reply:
Ok, that's interesting. The way to go in these cases is usually to change the `__class__` attribute, instead of copying `__dict__`. There are various reasons for this; you need to initialize the new class, and you lose the previous arguments you had set (here we lose `downcast_result.args`, check it out).

Originally, I assumed that the caller would do `result.click_result.<method>`, instead of `result.<method>`. It's a bit more verbose, but less magic as well. It's tests though, we can have magic code there, so we can go with your suggestion.